arch/xtensa/src/common/xtensa_initialstate.c

```diff
@@ -100,6 +124,71 @@ void up_initial_state(struct tcb_s *tcb)
   xcp->regs[REG_A1] = (uint32_t)tcb->stack_base_ptr + /* Physical top of stack frame    */
                                 tcb->adj_stack_size;
 
+  /* Each task access the TLS variables using the THREADPTR register plus an
+   * offset to obtain the address of the variable.
+   *
+   * TLS layout at link-time, where 0xNNN is the offset that the linker
+   * calculates to a particular TLS variable.
+   *
+   * LOW ADDRESS
+   *         |---------------------------|   Linker Symbols
+   *         | Section                   |   --------------
+   *         | .flash.rodata             |
+   *      0x0|---------------------------| <- _rodata_reserved_start
+   *       ^ | Other Data                |
+   *       | |---------------------------| <- _thread_local_start
+   *       | | .tbss                     | ^
+   *       V |                           | |
+   *   0xNNN | int example;              | | tls_area_size
+   *         |                           | |
+   *         | .tdata                    | V
+   *         |---------------------------| <- _thread_local_end
+   *         | Other data                |
+   *         | ...                       |
+   *         |---------------------------|
+   * HIGH ADDRESS
+   *
+   * Calculate the THREADPTR register's initialization value based on the
+   * link-time offset and the TLS area allocated on the stack.
+   *
+   * HIGH ADDRESS
+   *         |---------------------------| <- top of the Stack
+   *         |                           |
+   *         |                           |
+   *         |                           |
+   *         |                           |
+   *         |---------------------------|
+   *         | .tdata (*)                |
+   *       ^ | int example;              |
+   *       | |                           |
+   *       | | .tbss (*)                 |
+   *       | |---------------------------| <- start of TLS area
```

Comment:
is this what _thread_local_start points to?

Reply:
No, `_thread_local_start` refers to flash (and is used to calculate the offset `0xNNN`). Improving the documentation to make it clearer.